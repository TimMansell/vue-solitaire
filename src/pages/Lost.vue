<template>
  <GameOverlay
    center-content
    show-logo
    :is-visible="isVisible"
    data-test="lost"
  >
    <template #title> Sorry, no more Moves!</template>
    <template #msg>
      <GameSummary />
    </template>
    <template #buttons>
      <StartNewGameButton />
      <ShowHideBoardButton :is-show="isVisible" />
    </template>
  </GameOverlay>
</template>

<script>
import GameOverlay from '@/components/GameOverlay.vue';
import GameSummary from '@/components/GameSummary.vue';
import StartNewGameButton from '@/components/StartNewGameButton.vue';
import ShowHideBoardButton from '@/components/ShowHideBoardButton.vue';

export default {
  name: 'LostPage',
  components: {
    GameOverlay,
    StartNewGameButton,
    GameSummary,
    ShowHideBoardButton,
  },
  props: {
    show: {
      type: String,
      default: '',
    },
  },
  computed: {
    isVisible() {
      const { show } = this;

      return !show.includes('hide');
    },
  },
};
</script>
