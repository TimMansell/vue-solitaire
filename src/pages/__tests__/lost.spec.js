import { shallowMount } from '@vue/test-utils';
import Lost from '@/pages/Lost.vue';

describe('Lost.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(Lost);

    expect(wrapper.isVisible()).toBe(true);
  });
});
