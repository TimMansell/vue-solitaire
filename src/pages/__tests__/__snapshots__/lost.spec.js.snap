// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lost.vue matches snapshot 1`] = `
<gameoverlay-stub centercontent="true" showlogo="true" data-test="lost"> Sorry, no more Moves! <gamesummary-stub></gamesummary-stub>
  <startnewgamebutton-stub></startnewgamebutton-stub>
  <showboardbutton-stub></showboardbutton-stub>
</gameoverlay-stub>
`;
